### PR TITLE
Violin viz

### LINKED ISSUE #90: Create violin viz
to replace/add to the current greenness filtering

### PR DESCRIPTION
**Pull request by india-kerle**
---

# Description

This PR:
- creates a new notebook to generate an interactive violin plot per soc 2020 extension name
- add an aggregate_by_sic.py script to create one central dataframe aggregated by sic name
- updates the 2x2 typologies notebook to use the aggregated datasets
- updates the colors in the 2x2 typologies notebook to use the 'overall greenness' measure instead of the ad hoc greenness measure
- updates the ojo_analysis.yaml file to include the dates to use for the aggregated datasets so we're using the same datasets

Let's merge your chloropleth PR first; then merge this into the new-skills-viz branch and once those graphs are final, i can merge into dev.

Fixes # (issue)

This should close #90 

# Instructions for Reviewer

In order to test the code in this PR you need to ...

Please pay special attention to ...

# Checklist:

- [ ] I have refactored my code out from `notebooks/`
- [ ] I have checked the code runs
- [ ] I have tested the code
- [ ] I have run `pre-commit` and addressed any issues not automatically fixed
- [ ] I have merged any new changes from `dev`
- [ ] I have documented the code
  - [ ] Major functions have docstrings
  - [ ] Appropriate information has been added to `README`s
- [ ] I have explained this PR above
- [ ] I have requested a code review

### CODE REVIEW

File: dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py [hunk 1]
```diff
@@ -7,20 +7,16 @@
 
 import dap_prinz_green_jobs.analysis.ojo_analysis.process_ojo_green_measures as pg
 from dap_prinz_green_jobs.getters.data_getters import save_to_s3
-from dap_prinz_green_jobs import BUCKET_NAME, PROJECT_DIR, analysis_config
+from dap_prinz_green_jobs import BUCKET_NAME, analysis_config
 from dap_prinz_green_jobs.getters.ojo_getters import (
     get_mixed_ojo_location_sample,
     get_mixed_ojo_salaries_sample,
     get_large_ojo_location_sample,
-    get_large_ojo_salaries_sample,
```
Comment (lizgzil) [inline dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py]: the changes merged in `redochloropleths` will mean you shouldn't get rid of this. I guess you'll see when you resolve merge conflicts.

File: dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_sic.py [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+    # 2. extracted green skills
+    all_green_measures_df = pg.merge_green_measures(
+        skill_measures_df, occs_measures_df, inds_measures_df, soc_name_dict
+    )
+
+    if data_type == "mixed":
+        salary_information = get_mixed_ojo_salaries_sample()
+        locations_information = get_mixed_ojo_location_sample()
+    elif data_type == "large":
+        salary_information = get_large_ojo_location_sample()
```
Comment (lizgzil) [inline dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_sic.py]:
```suggestion
        salary_information = get_large_ojo_salaries_sample()
```
(and import it)

File: dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_soc.py [hunk 3]
```diff
@@ -12,18 +12,15 @@
     get_mixed_ojo_location_sample,
     get_mixed_ojo_salaries_sample,
     get_large_ojo_location_sample,
-    get_large_ojo_salaries_sample,
```
Comment (lizgzil) [inline dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_soc.py]: again, dont remove this

**Review by lizgzil — COMMENTED** [review]
Looks good! 🎻 
`dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_sic.py` ran and so did the notebook.

I commented about some bits about the salaries large sample that need to change. It's hard to comment on the notebook in the PR, but 2 nice to have things:
- The drop down isn't in alphabetical order - which would be nice
- I reckon get rid of the Number of ONET green topics violin since it might add confusion since its not tied to the greenness measures?

File: dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py (hunk 1 above)
Comment (india-kerle) [reply dap_prinz_green_jobs/analysis/ojo_analysis/aggregate_by_region.py]: okay no problems! i'll include it then

**Comment by india-kerle** [comment]
thanks so much for your review and great emoji choice @lizgzil - i've addressed your comments in the latest commit 🫡

**Pull request merged**